src/Components/ExtraInfo/index.jsx

```diff
@@ -0,0 +1,204 @@
+import React, { useEffect, useState } from 'react'
+import useAuth from '../../Context'
+import { Maps } from '../Maps'
+
+export const ExtraInfo = (props) => {
+  const [savedEventsArray, setsavedEventsArray] = useState([])
+  const { logout } = useAuth()
+  const { userSpotifyID } = useAuth()
+
+  useEffect(() => {
+    fetchSavedEvents()
+  }, [])
+
+  const fetchSavedEvents = async () => {
+    const result = await fetch(
+      `http://localhost:3001/events/${userSpotifyID}`,
+      {
+        method: 'GET',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+      }
+    )
+    const response = await result.json()
+    const eventIDs = response.map((event, index) => {
+      return event.event_id
+    })
+    console.log('eventIDs', eventIDs)
+    setsavedEventsArray(eventIDs)
+  }
+
+  const handleClick = async (eventID, event_name, date, city, url) => {
+    console.log('clicked')
+    if (savedEventsArray.includes(eventID)) {
+      try {
+        const data = {
+          event_id: eventID,
+        }
+        const result = await fetch(
+          `http://localhost:3001/events/${userSpotifyID}`,
+          {
+            method: 'DELETE',
+            headers: {
+              'Content-Type': 'application/json',
+            },
+            body: JSON.stringify(data),
+          }
+        )
+        console.log('savedEventsArray - removed', savedEventsArray)
+        fetchSavedEvents()
+      } catch (error) {
+        console.log('could not remove saved event')
+      }
+    } else {
+      try {
+        const data = {
+          event_name: event_name,
+          date: date,
+          city: city,
+          url: url,
+          event_id: eventID,
+        }
+        const result = await fetch(
+          `http://localhost:3001/events/${userSpotifyID}`,
+          {
+            method: 'POST',
+            headers: {
+              'Content-Type': 'application/json',
+            },
+            body: JSON.stringify(data),
+          }
+        )
+        const response = await result.json()
+        console.log('savedEventsArray - added', savedEventsArray)
+        fetchSavedEvents()
+      } catch (error) {
+        console.log('this event could not be saved', error)
```

Comment:
if things error it would be great to set some state and show something to the user that something went wrong

Reply:
added ticket to trello for this